compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1192,6 +1192,9 @@ struct RootCollector<'a, 'tcx> {
 
 impl<'v> RootCollector<'_, 'v> {
     fn process_item(&mut self, id: hir::ItemId) {
+        if self.tcx.hir().maybe_body_owned_by(id.def_id.def_id).is_some() {
```

Comment:
More seriously, you should add a comment about `-Zfuckit` (or a check for the flag, which I would prefer) here to explain why we need to lazily borrowck here.
(Also, this is another W for the query system :D)

Reply:
The comment is a good idea. I'm not sure it makes sense to gate `ensure()` under the flag though; the whole point of the query system is that calling the same function twice has no effect.